packages/react-storage/src/components/StorageImage/_tests_/StorageImage.test.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+import { render, screen } from '@testing-library/react';
+
+import { Storage } from 'aws-amplify';
+import { ComponentClassNames } from '@aws-amplify/ui-react';
+
+import { StorageImage } from '../StorageImage';
+
+describe('StorageImage:', () => {
+  const imgKey = 'test.jpg';
+  const imgURL = 'https://amplify.s3.amazonaws.com/path/to/test.jpg';
+  const fallbackSrc = 'https://amplify.s3.amazonaws.com/path/to/fallback.jpg';
+  const errorMessage = '500 Internal Server Error';
+
+  beforeAll(() => {
+    jest.spyOn(Storage, 'get').mockResolvedValue(imgURL);
+  });
+
+  it('should render classname', async () => {
+    render(
+      <StorageImage alt="StorageImage" imgKey={imgKey} accessLevel="public" />
+    );
+
+    const img = await screen.findByRole('img');
+    expect(img).toHaveClass(ComponentClassNames.StorageImage);
+  });
+
+  it('should set loading attribute to lazy if isLazy prop is true', async () => {
+    render(
+      <StorageImage
+        alt="StorageImage"
+        imgKey={imgKey}
+        accessLevel="public"
+        isLazy
+      />
+    );
+
+    const img = await screen.findByRole('img');
+    expect(img).toHaveAttribute('loading', 'lazy');
+  });
+
+  it('should get the presigned URL and pass it to image src attribute', async () => {
+    const onStorageError = jest.fn();
+    render(
+      <StorageImage
+        alt="StorageImage"
+        imgKey={imgKey}
+        accessLevel="public"
+        onStorageError={onStorageError}
+      />
+    );
+
+    const img = await screen.findByRole('img');
+    expect(onStorageError).not.toHaveBeenCalled();
+    expect(img).toHaveAttribute('src', imgURL);
+  });
+
+  it('should set image src attribute to fallbackSrc and invoke onStorageError when Storage get is rejected', async () => {
```

Comment:
Future readers may not have immediate context in to "get" being a method of `Storage`:
```suggestion

  it('should set image src attribute to fallbackSrc and invoke onStorageError when Storage.get is rejected', async () => {
```